controllers/front/SmailyRssFeed.php

```diff
@@ -32,7 +32,33 @@ public function initContent()
 
     public function generateRssFeed()
     {
-        $products = Product::getProducts($this->context->language->id, 0, 50, 'date_upd', 'desc', false, true);
+        // Hardcoded values in v1.3.0 and older.
+        $limit = 50;
+        $order_by = 'date_upd';
+        $order_way = 'desc';
+        $id_category = false;
+        // Make sure the values are what we are expecting.
+        if (Tools::getValue('limit') >= 1 && is_int(Tools::getValue('limit'))) {
+            $limit = Tools::getValue('limit');
+        }
+        if (in_array(Tools::getValue('order_by'), array('date_add', 'date_upd', 'name', 'price', 'id_product'))) {
+            $order_by = Tools::getValue('order_by');
+        }
+        if (Tools::getValue('order_way') === 'asc' || Tools::getValue('order_way') === 'desc') {
+            $order_way = Tools::getValue('order_way');
+        }
+        if (Tools::getValue('id_category') && Tools::getValue('id_category') === 'all_products' || is_int(intval(Tools::getValue('id_category')))) {
+            $id_category = Tools::getValue('id_category');
+        }
```

Comment:
Calling exact same function multiple times is not the best practice to get used to for 2 reasons:
- depending on the function exact same call can yield in different results (especially when querying the database);
- it can be taxing as the function can execute heavy/time consuming operations;

I would rather see you fetching the required state once and then use it where needed in the function.

Perhaps you could rework the conditional logic to something like:
```suggestion
        $limit = (int) Tools::getValue('limit');
        $limit = $limit >= 1 ? $limit : 50;
        
        $allowed_order_by_values = array('date_add', 'date_upd', 'name', 'price', 'id_product');  // This could be converted to a class variable so you can reuse it when validating configuration values during update.
        $order_by = Tools::getValue('order_by');
        $order_by = in_array($order_by, $allowed_order_by_values, true) ? $order_by : 'date_upd';
        
        $order_way = Tools::getValue('order_way');
        $order_way = in_array($order_way, array('asc', 'desc'), true) ? $order_way : 'desc';
        
        $id_category = Tools::getValue('id_category');
        if ($id_category !== 'all_products' && (int) $id_category <= 0) {
                $id_category = false;  // Though, shouldn't this be "all_products"?
        }
```

Reply:
Fetching it into a variable once and comparing now